site/app/models/gradeable/GradedGradeable.php

```diff
@@ -69,6 +68,19 @@ public function setTaGradedGradeable(TaGradedGradeable $ta_graded_gradeable) {
         $this->ta_graded_gradeable = $ta_graded_gradeable;
     }
 
+    /**
+     * Gets the TaGradedGradeable for this graded gradeable, or generates a blank
+     *  one if none exists and $generate is true
+     * @param bool $generate If a new TaGradedGradeable should be generated if one doesn't exist already
+     * @return TaGradedGradeable
+     */
+    public function getTaGradedGradeable($generate = false) {
+        if ($this->ta_graded_gradeable === null && $generate === true) {
+            $this->ta_graded_gradeable = new TaGradedGradeable($this->core, $this, []);
+        }
```

Comment:
Another getter that is doubling as a setter. I would just have a `createTaGradedGradeable` function if you really don't want anyone else touching these classes.